staging/src/k8s.io/apimachinery/pkg/fields/selector.go

```diff
@@ -116,6 +116,9 @@ func (t *hasTerm) Requirements() Requirements {
 }
 
 func (t *hasTerm) String() string {
+	if t == nil {
```

Comment:
What if we omit the selector changes completely? There's no good reason to be calling String() on nil terms at all, and I'm nervous about what happens if there is a nil one and it's nested in some weird way.

Reply:
Ok, I have reverted change about `hasTerm` and `notHasTerm`